### PR TITLE
Add contextual information to MDC for audit logs

### PR DESCRIPTION
**Pull request by Sathiyakugan**
Resolves wso2/product-is#9119

**Description**
Added contextual information such as logged-in user, SP name, the remote IP address. Client user agent using MDC.

### CODE REVIEW

File: components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/OAuth2AccessTokenHandler.java [hunk 1]
```diff
@@ -124,6 +126,17 @@ protected AuthenticationResult doAuthenticate(MessageContext messageContext) {
...
                         AuthConfigurationUtil.getInstance().isScopeValidationEnabled());
+                try {
+                    if (MDC.get(SERVICE_PROVIDER) != null) {
+                        MDC.put(SERVICE_PROVIDER,
+                                OAuth2Util.getServiceProvider(clientApplicationDTO.getConsumerKey())
+                                        .getApplicationName());
+                    }
+                } catch (IdentityOAuth2Exception e) {
+                    if (log.isDebugEnabled()) {
+                        log.error("Error occurred while getting the Service Provider by Consumer key");
```
Comment (vihanga-liyanage) [inline components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/OAuth2AccessTokenHandler.java:137]: Fix this please

File: components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/OAuth2AccessTokenHandler.java [hunk 2]
```diff
@@ -124,6 +126,15 @@ protected AuthenticationResult doAuthenticate(MessageContext messageContext) {
                 authenticationContext.addParameter(OAUTH2_ALLOWED_SCOPES, responseDTO.getScope());
                 authenticationContext.addParameter(OAUTH2_VALIDATE_SCOPE,
                         AuthConfigurationUtil.getInstance().isScopeValidationEnabled());
+                try {
+                    if (MDC.get(SERVICE_PROVIDER) != null) {
```
Comment (ayshsandu) [inline components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/OAuth2AccessTokenHandler.java:130]: From where the SERVICE_PROVIDER is set to MDC initially. If this is the only place, MDC.put(SERVICE_PROVIDER.. will never happen.

Comment (Sathiyakugan) [reply components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/OAuth2AccessTokenHandler.java:130]: Already fixed https://github.com/wso2-extensions/identity-carbon-auth-rest/blob/master/components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/OAuth2AccessTokenHandler.java#L152

File: components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/OAuth2AccessTokenHandler.java [hunk 3]
```diff
@@ -124,6 +126,15 @@ protected AuthenticationResult doAuthenticate(MessageContext messageContext) {
...
+                        MDC.put(SERVICE_PROVIDER,
+                                OAuth2Util.getServiceProvider(clientApplicationDTO.getConsumerKey())
+                                        .getApplicationName());
+                    }
+                } catch (IdentityOAuth2Exception e) {
+                    log.error("Error occurred while getting the Service Provider by Consumer key");
```
Comment (hasinthaindrajee) [inline components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/OAuth2AccessTokenHandler.java:136]: Why do we need to swallow the stacktrace here. Don't we loose important information here  ?

Comment (Sathiyakugan) [reply components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/OAuth2AccessTokenHandler.java:136]: Fixed with this pr https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/134

**Pull request merged**